kernel/init.c

```diff
@@ -212,6 +212,19 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 	static const unsigned int boot_delay;
 #endif
 
+#ifdef CONFIG_INIT_STACKS
+	/* Disable local interrupts on this CPU. Other CPUs haven't been
+	 * started yet.
+	 */
+	int key = arch_irq_lock();
```

Comment:
Does it make sense to add the lock-unlock into CONFIG_SMP, then, and save a few bytes of code space for non-SMP builds?

Reply:
I don't follow you here.
It's essential that local interrupts are not delivered while we are setting the irq stack buffer.
This doesn't have anything to do with SMP.

Reply:
Since interrupts are enabled at this point, we must lock-unlock.

Reply:
Right, my bad, overlooked that IRQs are enabled at bg_thread_main